src/main/io/gps_nmea.c

```diff
@@ -254,6 +255,58 @@ static bool gpsReceiveData(void)
     return hasNewData;
 }
 
+#ifdef GPS_PROTO_MTK
+
+static uint8_t nmea_conf0[]="$PMTK251,57600*2C\r\n"; //change baudrate to 57600
+static uint8_t nmea_conf1[]="$PMTK314,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0*28\r\n"; //disable all messages except GGA and RMC
+static uint8_t nmea_conf2[]="$PMTK220,200*2C\r\n"; //5Hz update, should works for most modules
+static uint8_t nmea_conf3[]="$PMTK220,100*2F\r\n"; //try set 10Hz update if supported
+
+// Send NMEA command like normal string
+static bool nmeaTransmitAutoConfigCommands(const uint8_t * cmd) {
+    while (serialTxBytesFree(gpsState.gpsPort) > 0) {
+        if (cmd[gpsState.autoConfigPosition] != 0) {
+            serialWrite(gpsState.gpsPort, cmd[gpsState.autoConfigPosition]);
+            gpsState.autoConfigPosition++;
+        }
+        else if (isSerialTransmitBufferEmpty(gpsState.gpsPort)) {
+            gpsState.autoConfigStep++;
+            gpsState.autoConfigPosition = 0;
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    return false;
+}
+
+bool gpsConfigure(void)
+{
+    switch (gpsState.autoConfigStep) {
```

Comment:
Considering the above comment this can be implemented simpler:
```
if (gpsState.autoConfigStep < sizeof(mtk_conf)/sizeof(mtk_conf[0]) {
    nmeaTransmitAutoConfigCommands(mtk_conf[gpsState.autoConfigStep]);
}
else {
    gpsSetState(GPS_RECEIVING_DATA);
}
```